haystack_experimental/components/extractors/llm_document_content_extractor.py

```diff
@@ -0,0 +1,243 @@
+        if prompt is None:
+            return {"error": "Document has no content, skipping LLM call."}
+
+        try:
+            result = self._chat_generator.run(messages=[prompt])
+        except Exception as e:
+            if self.raise_on_failure:
+                raise e
+            logger.error(
+                "LLM {class_name} execution failed. Skipping metadata extraction. Failed with exception '{error}'.",
+                class_name=self._chat_generator.__class__.__name__,
+                error=e,
+            )
+            result = {"error": "LLM failed with exception: " + str(e)}
+        return result
+
+    @component.output_types(documents=List[Document], failed_documents=List[Document])
+    def run(self, documents: List[Document]) -> Dict[str, List[Document]]:
+        """
+        Run content extraction on a list of image-based documents using a vision-capable LLM.
+
+        Each document is passed to the LLM along with a predefined prompt. The response is used to update the document's
+        content. If the extraction fails, the document is returned in the `failed_documents` list with metadata
+        describing the failure.
+
+        :param documents: A list of image-based documents to process. Each must have a valid file path in its metadata.
+        :returns:
+            A dictionary with:
+            - "documents": Successfully processed documents, updated with extracted content.
+            - "failed_documents": Documents that failed processing, annotated with failure metadata.
+        """
+        if not documents:
+            return {"documents": [], "failed_documents": []}
+
+        # Create ChatMessage prompts for each document
+        image_contents = self._document_to_image_content.run(documents=documents)["image_contents"]
+        all_prompts: List[Union[ChatMessage, None]] = []
+        for image_content in image_contents:
+            if image_content is None:
+                # If the image content is None, it means the document could not be converted to an image.
+                # We skip this document.
+                # We don't log a warning here since it is already logged in the DocumentToImageContent component.
+                all_prompts.append(None)
+                continue
+            message = ChatMessage.from_user(content_parts=[TextContent(text=self.prompt), image_content])
+            all_prompts.append(message)
+
+        # Run the LLM on each prompt
+        with ThreadPoolExecutor(max_workers=self.max_workers) as executor:
+            results = executor.map(self._run_on_thread, all_prompts)
+
+        successful_documents = []
+        failed_documents = []
+        for document, result in zip(documents, results):
+            if "error" in result:
+                new_meta = {
+                    **document.meta,
+                    "content_extraction_error": result["error"],
+                    "content_extraction_response": None,
```

Comment:
this field seems unused. Right?

Reply:
good point! This would be safe to remove